### ISSUE TITLE
UX Enhancement: Assigning exsisting IP to interface

### ISSUE THREAD

**Issue by kotso** [netbox-community/netbox#15147]
### NetBox version

v.3.7.2

### Feature type

New functionality

### Proposed functionality

When adding existing IP to interface, after clicking add icon, you have option to search for IP/VRF to find desired IP, in list of found IP's you have only option to navigate to IP edit page and set interface from there.

So it's even easier to just look for IP and set interface instead of navigating to interface page and clicking add.

I suggest that list of found IP addresses should just have button "Assign" that would assign them to interface we are currently editing.

### Use case

current flow is IMHO not logical, as after finding address you are starting same assign operation from scratch.

### Database changes

No database change required

**Comment by sleepinggenius2** [comment]
On the "Assign IP" tab, after you search for an existing IP address, if you click on the link in one of the search results, it does take you to the edit page for that IP, but it also assigns the interface that you were on. Sure, it's one extra click at that point to click on the "Save" button, but I find that I often need to change an attribute on the IP anyway, like changing the status from Reserved to Active, so it actually saves time in the long run.

**Comment by rkiss** [comment]
I was about to propose the same enhancement. I would go a bit further than the original ask and add a popup with two actions inside: "Assign" and "Assign and make primary". When selected it shall just do that, no question ask, and then go back to the corresponding interfaces tab of the device.
If the IP is already assigned to something else it should ask for a confirmation perhaps.
![image](https://github.com/netbox-community/netbox/assets/5818049/cb072a94-2cef-441a-aad6-be134b452d94)

**Comment by sleepinggenius2** [comment]
I think there is some confusion about the process here. You should be clicking on the IP link over on the left, like where it says 10.241.168.1/23 or 10.241.168.10/23 in your screenshot. That will drop you into the edit screen for that IP **with the selected interface assignment pre-populated**. Then you just need to hit "Save" or make any other changes you need first, like selecting "Make this the primary IP for the device/VM", as requested, or changing the status from Reserved to Active. It will then send you back to the interfaces list for the device where you came from. It sounds like the existing functionality is already sufficient to meet all the requested feature enhancements.

**Comment by rkiss** [comment]
I don't think that there is any confusion here. What you are describing is what we are all doing. The proposal is to make the process slightly faster, more streamlined.
I found that while working with Netbox there are many things where the number of clicks and waits for pages to load can be reduced if the right actions are available at a higher level. This is one of them.

**Comment by sleepinggenius2** [comment]
> current flow is IMHO not logical, as after finding address you are starting same assign operation from scratch.

While you may be following the correct process, the way that the use case above from the original feature request was written, it does not sound like that was the process that OP was following, but instead clicking the yellow edit button. Therefore, that is what I was basing my comments on. If your objective is to add a feature to reduce the one extra page load and click from how it is today, then that seems like a different use case than the feature request proposed here.

As an aside, we would never use such a feature in our instance, as our business process requires a number of things to be done with an IP address between reservation and assignment, like changing the status and generally validating things like the role and tenant, so the current process already meets our needs very well.

**Comment by jeffgdotorg** [comment]
Thanks for surfacing this UX gap. This kind of "papercut" struggle really adds up, and we could address it with a modest effort. I'm marking this issue `needs owner`; if you're interested in working it to a PR, please comment and somebody will assign it to you. Otherwise anybody with the right skills and spare working capacity can pick it up.
